packages/analytics-core/src/plugins/destination.ts

```diff
@@ -271,55 +252,37 @@ export class Destination implements DestinationPlugin {
     this.config.loggerProvider.warn(getResponseBodyString(res));
 
     this.config.flushQueueSize /= 2;
-
-    const tryable = this.getTryableList(list);
-    this.scheduleTryable(tryable);
+    // All the events are still queued, and will be retried with next trigger
   }
 
   handleRateLimitResponse(res: RateLimitResponse, list: Context[]) {
     const dropUserIds = Object.keys(res.body.exceededDailyQuotaUsers);
     const dropDeviceIds = Object.keys(res.body.exceededDailyQuotaDevices);
-    const throttledIndex = res.body.throttledEvents;
     const dropUserIdsSet = new Set(dropUserIds);
     const dropDeviceIdsSet = new Set(dropDeviceIds);
-    const throttledIndexSet = new Set(throttledIndex);
 
-    const retry = list.filter((context, index) => {
+    const retry = list.filter((context) => {
       if (
         (context.event.user_id && dropUserIdsSet.has(context.event.user_id)) ||
         (context.event.device_id && dropDeviceIdsSet.has(context.event.device_id))
       ) {
         this.fulfillRequest([context], res.statusCode, res.body.error);
         return;
       }
-      if (throttledIndexSet.has(index)) {
-        context.timeout = this.throttleTimeout;
-      }
```

Comment:
When 429, we should wait for 30 seconds before retry according to the [docs](https://www.docs.developers.amplitude.com/analytics/apis/http-v2-api/#429-response-toomanyrequestsfordeviceerror). If we want to enforce event order, we can just pause sending for the all events. Simply deleting it 
 would keep getting 429.

Reply:
Thanks for brining this up. I also find that. However, pausing sending events for 30 minutes will blow the storage if there are a lot of requests per second, isn't it?  

This is how other SDK behave. 
old android -> do nothing
new android -> split file, triggerBackOff (increase the flush interval) -> this is wrong.
new ios -> di nothing
old javascript -> do nothing.

I make it align with others behavior.

Reply:
Why pausing sending events for 30 **seconds** will blow the storage? 

> This is how other SDK behave.
old android -> do nothing
new android -> split file, triggerBackOff (increase the flush interval) -> this is wrong.
new ios -> di nothing
old javascript -> do nothing.

Amplitude-Swift just wait for the next flush [here](https://github.com/amplitude/Amplitude-Swift/blob/0cd014a332d098e70f760dfc2b0cb75f5ef74abe/Sources/Amplitude/Utilities/PersistentStorageResponseHandler.swift#L113-L114). But I do think this is a bug. @crleona wdyt?

Reply:
So on a 429 the legacy iOS SDK does [exponential backoff](https://github.com/amplitude/Amplitude-iOS/blob/bfdec453a31fd35942a619ef472fcf2f09e2313a/Sources/Amplitude/Amplitude.m#L1172) on a 429, the new Android SDK looks like it just [drops](https://github.com/amplitude/Amplitude-Kotlin/blob/5c155e2b7c0c6b55ab9870469ce4a05e79275ce8/core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt#L122) the file (which is definitely bad!).

I'm definitely not concerned with increasing storage size too much if we do backoff, we're probably talking thousands of events to get past a few MB in storage, which will easily fit in memory or a file.

That said, I'd still expect us to make forward progress uploading events without backoff as a rejected request shouldn't count toward our quota (though, our specific implementation may not match this assumption). We'd just expect more retries until one eventually goes through.

I would recommend leaving this in though (perhaps even moving to an exponential backoff model). It's a defense against misconfigured clients (perhaps with a too small flush interval), and it's not like an earlier request is going to do anything but 429 again.

Reply:
Thanks for the feedback. Align offline. We determine to hold 30 seconds. I will add this in a separate PR.

Reply:
Btw, we don't need to wait for 30 seconds before sending events again. Our default EPS is 30 events per second. So if sending an event fails, trying again within 1 second should be sufficient.

Reply:
nvm. just read the doc that Xinyi shared.

Reply:
Added the rate limit handler in this [pr](https://github.com/amplitude/Amplitude-TypeScript/pull/754).